app/controllers/alchemy/pages_controller.rb

```diff
@@ -79,15 +79,15 @@ def render_page_or_redirect
         redirect_legacy_page
       elsif @page.blank?
         page_not_found!
-      elsif multi_language? && params[:locale].blank?
+      elsif multi_language? && params[:locale].blank? && !default_locale?
         redirect_page(locale: Language.current.code)
-      elsif multi_language? && params[:urlname].blank? && !params[:locale].blank? && configuration(:redirect_index)
+      elsif multi_language? && params[:urlname].blank? && params[:locale].present? && configuration(:redirect_index)
```

Comment:
Metrics/LineLength: Line is too long. [116/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)